test/socket/test_socket.rb

```diff
@@ -605,6 +605,116 @@ def test_connect_timeout
     sock.close if sock && ! sock.closed?
   end
 
+  def test_getaddrinfo_v4_slow
+    original = Addrinfo.singleton_method(:getaddrinfo)
+    begin
+      server = TCPServer.new("::1", 0)
+    rescue Errno::EADDRNOTAVAIL # IPv6 is not supported
+      return
+    end
+    port = server.addr[1]
+    Addrinfo.define_singleton_method(:getaddrinfo) do |_, _, family, _|
+      if family == :PF_INET
+        sleep(10)
+        [Addrinfo.tcp("127.0.0.1", port)]
+      else
+        [Addrinfo.tcp("::1", port)]
+      end
+    end
+    serv_thread = Thread.new { server.accept }
+    sock = Socket.tcp("localhost", port)
+    accepted = serv_thread.value
+  ensure
+    Addrinfo.define_singleton_method(:getaddrinfo, original)
```

Comment:
`assert_separately` would be useful to separate modified environment in a different process.

Reply:
Thank you.
`assert_separately` is useful in this case.